code/__HELPERS/unsorted.dm

```diff
@@ -721,11 +721,12 @@ Turf and target are seperate in case you want to teleport some distance from a t
 	var/pixel_y_offset = AM.pixel_y + M.get_y_shift()
 
 	//Irregular objects
-	if(AM.bound_height != world.icon_size || AM.bound_width != world.icon_size)
-		var/icon/AMicon = icon(AM.icon, AM.icon_state)
+	var/icon/AMicon = icon(AM.icon, AM.icon_state)
+	var/icon/AMiconheight = AMicon.Height()
```

Comment:
Which is faster, object or global procs? or are they about the same?
Because: ```Height()``` and ```Width()``` just call ```_icon_height(icon)``` and ```_icon_width(icon)``` (might need to check the args in stdlib)

Granted, most of the cost here comes from making the Icon at all, they suck royally.

also these shouldn't be pathed ```/icon```, they're numbers.